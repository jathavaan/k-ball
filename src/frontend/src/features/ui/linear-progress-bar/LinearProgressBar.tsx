import { LinearProgressProps } from "@mui/material";
import { StyledLinearProgressBar } from "@features/ui/linear-progress-bar/linearProgressBar.style.ts";

export const LinearProgressBar = (props: LinearProgressProps) => {
  return (
    <StyledLinearProgressBar data-testid="linear-progress-bar" {...props} />
  );
};
