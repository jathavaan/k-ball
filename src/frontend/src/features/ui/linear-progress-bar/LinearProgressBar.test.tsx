import { render, screen } from "@testing-library/react";
import { LinearProgressBar } from "./LinearProgressBar";

describe("Tests for LinearProgressBar", () => {
  it("Should render the LinearProgressBar", () => {
    render(<LinearProgressBar />);

    const progressBarElement = screen.getByTestId("linear-progress-bar");
    expect(progressBarElement).toBeDefined();
  });

  it("Should match LinearProgressBar snapshot", () => {
    const { asFragment } = render(<LinearProgressBar />);
    expect(asFragment()).toMatchSnapshot();
  });
});
