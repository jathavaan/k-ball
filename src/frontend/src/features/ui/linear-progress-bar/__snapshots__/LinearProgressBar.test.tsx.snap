// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LinearProgressBar > Should match LinearProgressBar snapshot 1`] = `
<DocumentFragment>
  <span
    class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-indeterminate css-a9j3vj-MuiLinearProgress-root"
    data-testid="linear-progress-bar"
    role="progressbar"
  >
    <span
      class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate css-eg0r9f-MuiLinearProgress-bar1"
    />
    <span
      class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar2Indeterminate css-1yf7653-MuiLinearProgress-bar2"
    />
  </span>
</DocumentFragment>
`;

exports[`LinearProgressBar > Should match LinearProgressBar snapshot 1`] = `
<DocumentFragment>
  <span
    class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-indeterminate css-a9j3vj-MuiLinearProgress-root"
    data-testid="linear-progress-bar"
    role="progressbar"
  >
    <span
      class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate css-eg0r9f-MuiLinearProgress-bar1"
    />
    <span
      class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar2Indeterminate css-1yf7653-MuiLinearProgress-bar2"
    />
  </span>
</DocumentFragment>
`;
